gatsby-config.ts

```diff
@@ -230,9 +230,19 @@ const config: GatsbyConfig = {
       },
     },
     // Needed for Gatsby Cloud redirect support
-    `gatsby-plugin-gatsby-cloud`,
+    {
+      resolve: `gatsby-plugin-gatsby-cloud`,
+      options: {
+        generateMatchPathRewrites: false,
+      },
+    },
     // Creates `_redirects` & `_headers` build files for Netlify
-    `gatsby-plugin-netlify`,
+    {
+      resolve: `gatsby-plugin-netlify`,
+      options: {
+        generateMatchPathRewrites: false,
```

Comment:
ref https://www.gatsbyjs.com/plugins/gatsby-plugin-gatsby-cloud/#configuration

Reply:
A bit more context on this. Our i18n plugin is generating these `matchPath` routes for each locale, for client side routing purposes:
```js
if (newPage.path.match(/^\/[a-z]{2}\/404\/$/)) {
  // Match all paths starting with this code (apart from other valid paths)
  newPage.matchPath = `/${locale.code}/*`
}
```

If this flag is on, it would write in the `_redirects` file, something like:
```
/en/*  /en/404  200
/es/*  /es/404  200
```

Note the 200 status code. We don't want that. That will create a redirect. We want a 404 status code in there. That is why we are disabling this and creating those rules ourselves.